fauna/models.py

```diff
@@ -16,43 +16,82 @@ def __hash__(self):
         hash(self.name)
 
 
-class DocumentReference:
-    collection: Module
-    ref_id: str
+class BaseReference:
+    _collection: Module
+    _id: str
 
-    def __init__(self, collection: Union[str, Module], ref_id: str):
-        if isinstance(collection, str):
-            self.collection = Module(collection)
+    def __init__(self, coll: Union[str, Module], id_: str):
+        if isinstance(coll, str):
+            self._collection = Module(coll)
         else:
-            self.collection = collection
+            self._collection = coll
 
-        self.ref_id = ref_id
+        self._id = id_
 
     def __str__(self):
-        return f"{self.collection}:{self.ref_id}"
+        return f"{self._collection}:{self._id}"
 
     def __eq__(self, other):
-        return isinstance(other, DocumentReference) and str(self) == str(other)
+        return isinstance(other, type(self)) and str(self) == str(other)
 
     def __hash__(self):
-        hash((self.collection, self.ref_id))
+        hash((self._collection, self._id))
+
+    @property
+    def collection(self) -> Module:
```

Comment:
My instinct is to keep the shorter `coll` name since that is how people will interact with the field within fql.

Reply:
I have an aversion to abbreviations, but if you think it will confuse people to use collection, I'll go with coll.

Reply:
> I have a bit of an aversion to abbreviations
Oh, me too! But in this case I think the consistency with core is more important.